### DISCUSSION TITLE
Zero gas fee on polygon edge sidechain?

### DISCUSSION THREAD

**Discussion by taylorkchan** [0xPolygon/polygon-edge#577] [General] [answered]
Hi, our team has been testing transactions on our sidechain, it seems the gas fee is always zero. Can the team confirm this is the case? There is no need of paying gas fee in ethers in any transactions?

**Comment by zivkovicmilos** [comment, marked as answer]
Hey @taylorkchan,

The default price limit that nodes enforce is `0`, meaning the nodes will accept transactions that have a gas price set to `0`.

If you want to set a lower limit on the gas price transactions need to have (and thus users need to pay), you can check out the `--price-limit` flag on the `server` command, that's what you're looking for.

**Reply by taylorkchan** [reply]
Thanks! this is what i am looking for.

**Reply by sachinjain551836** [reply]
What would be the exact command run for setting price limit and where to run? should be inside the node folder or chain folder, I tried but not able to set the value. 
and which file contains this setting after run the said command
polygon-edge server --price-limit 1000
Can you please share screenshot..

**Reply by taylorkchan** [reply]
You need to set it on the genisis file, see step 3 https://docs.polygon.technology/docs/edge/get-started/set-up-ibft-on-the-cloud

**Reply by sachinjain551836** [reply]
I tried but getting error.
<img width="948" alt="Screenshot 2022-08-03 090651" src="https://user-images.githubusercontent.com/109588784/182518726-147a228b-798e-42a9-b279-978f5cc184ca.png">

I run this cmmand

polygon-edge genesis --name RH-Chain --pos --consensus ibft --ibft-validators-prefix-path test-chain- --bootnode /ip4/127.0.0.1/tcp/10001/p2p/16Uiu2HAm33eLaSCRwWFLzuaEPRQMDeSmb5rcUvT5yyPyPv1mfKht --bootnode /ip4/127.0.0.1/tcp/20001/p2p/16Uiu2HAmSVAkztQnhFSLZNBypzZGX7d5R3tAVCMKqE8yL4AjQvVX --premine=0x1477cc2adBbDd9a4b3adb37Bc9575EE0732B5000:10000000000000000000000000 server --price-limit 1000

**Reply by sachinjain551836** [reply]
Always getting "Error: unknown flag: --price-limit"

I tried : 
polygon-edge genesis --name RH-Chain server --price-limit 1000 --pos --consensus ibft --ibft-validators-prefix-path test-chain- --bootnode /ip4/127.0.0.1/tcp/10001/p2p/16Uiu2HAm33eLaSCRwWFLzuaEPRQMDeSmb5rcUvT5yyPyPv1mfKht --bootnode /ip4/127.0.0.1/tcp/20001/p2p/16Uiu2HAmSVAkztQnhFSLZNBypzZGX7d5R3tAVCMKqE8yL4AjQvVX --premine=0x1477cc2adBbDd9a4b3adb37Bc9575EE0732B5000:10000000000000000000000000

**Reply by laviniat1996** [reply]
@sachinjain551836 Sorry for the delayed reply. The `--price-limit` flag is a server flag, you don't have to include it in the genesis file.  You will have to add it when starting a node. An example: `go run main.go server --data-dir ./test-chain-1 --chain genesis.json --grpc-address :10000 --libp2p :10001 --jsonrpc :10002 --seal --price-limit 1000000000`. This sets the price limit as "1000000000" for the node.
